### ISSUE TITLE
Refactor the server to be contained in a struct

### ISSUE THREAD

**Issue by rayhaanj** [netsec-ethz/rains#87]
The server should be in the form of a struct which has configuration parameters in it and the relevant listen methods implemented on that struct. This would eliminate all the package local variables which are referenced from a lot of places, but are actually defined in many files making it hard to follow what's going on.

This would also clean up issue #36.

**Comment by fehlmach** [comment]
Done
